range_sensor_layer/src/range_sensor_layer.cpp

```diff
@@ -376,7 +397,21 @@ void RangeSensorLayer::update_cell(double ox, double oy, double ot, double r, do
     ROS_DEBUG("%f %f | %f %f = %f", dx, dy, theta, phi, sensor);
     ROS_DEBUG("%f | %f %f | %f", prior, prob_occ, prob_not, new_prob);
     unsigned char c = to_cost(new_prob);
+
     setCost(x, y, c);
+    if(use_decay_)
+    {
+      std::pair<int, int> coordinate_pair(x, y);
```

Comment:
Can you please add a comment to this section to explain what is going on (particularly with the else if )